src/modules/admin/components/Tokens/TokensMoveDialogForm.tsx

```diff
@@ -0,0 +1,320 @@
+    [colonyAddress],
+    [colonyAddress],
+  );
+
+  // Map the colony's domains to Select options
+  const domainOptions = useMemo(
+    () => [
+      { value: 1, label: 'root' },
+      ...(domains || []).map(({ id, name }) => ({ value: id, label: name })),
+    ],
+    [domains],
+  );
+
+  // Get from and to domain permissions for current user
+  const currentUserAddress = useSelector(walletAddressSelector);
+  const {
+    data: fromDomainPermissions,
+    isFetching: isFetchingFromDomainPermissions,
+  } = useUserDomainRoles(
+    colonyAddress,
+    values.fromDomain || 1,
+    currentUserAddress,
+    true,
+  );
+  const {
+    data: toDomainPermissions,
+    isFetching: isFetchingToDomainPermissions,
+  } = useUserDomainRoles(
+    colonyAddress,
+    values.toDomain || 1,
+    currentUserAddress,
+    true,
+  );
+
+  // Get domain token balances from state
+  const fromDomainTokenBalanceSelector = useCallback(
+    state =>
+      tokenBalanceSelector(
+        state,
+        colonyAddress,
+        values.tokenAddress || '',
+        values.fromDomain || 1,
+      ),
+    [colonyAddress, values.fromDomain, values.tokenAddress],
+  );
+  const fromDomainTokenBalance = useMappedState(fromDomainTokenBalanceSelector);
+  const toDomainTokenBalanceSelector = useCallback(
+    state =>
+      tokenBalanceSelector(
+        state,
+        colonyAddress,
+        values.tokenAddress || '',
+        values.toDomain || 1,
+      ),
+    [colonyAddress, values.toDomain, values.tokenAddress],
+  );
+  const toDomainTokenBalance = useMappedState(toDomainTokenBalanceSelector);
+
+  // Perform form validations
+  useEffect(() => {
+    if (
+      fromDomainTokenBalance &&
+      fromDomainTokenBalance.lt(new BigNumber(values.amount))
+    ) {
+      setErrors({ amount: MSG.noBalance });
+    }
+
+    if (
+      values.fromDomain &&
+      !isFetchingFromDomainPermissions &&
+      !fromDomainPermissions[COLONY_ROLE_FUNDING]
+    ) {
+      setErrors({ fromDomain: MSG.noPermissionFrom });
+    }
+
+    if (
+      values.toDomain &&
+      !isFetchingToDomainPermissions &&
+      !toDomainPermissions[COLONY_ROLE_FUNDING]
+    ) {
+      setErrors({ toDomain: MSG.noPermissionTo });
+    }
+
+    if (
+      values.toDomain !== undefined &&
+      values.toDomain === values.fromDomain
+    ) {
+      setErrors({ toDomain: MSG.samePot });
+    }
+  }, [
+    fromDomainPermissions,
+    fromDomainTokenBalance,
+    isFetchingFromDomainPermissions,
+    isFetchingToDomainPermissions,
+    selectedTokenRef,
+    setErrors,
+    toDomainPermissions,
+    values.amount,
+    values.fromDomain,
+    values.toDomain,
+  ]);
+
+  return (
+    <>
+      <FormStatus status={status} />
+      <DialogSection>
+        <Heading
+          appearance={{ size: 'medium', margin: 'none' }}
+          text={MSG.title}
+        />
+      </DialogSection>
+      <DialogSection>
+        <Select options={domainOptions} label={MSG.from} name="fromDomain" />
```

Comment:
Can we better style the error message here? As it is it looks kinda broken...

![Screenshot from 2019-09-25 09-17-22](https://user-images.githubusercontent.com/1193222/65575424-71cf6e80-df78-11e9-9757-8eb851564ab2.png)

Reply:
Nice catch, thanks. Correctly set the `appearance.theme` now 👍